compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/PointerInputMatcher.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation
+
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.input.pointer.PointerButton
+import androidx.compose.ui.input.pointer.PointerEvent
+import androidx.compose.ui.input.pointer.PointerType
+import androidx.compose.ui.util.fastAll
+
+/**
+ * [PointerInputMatcher] represents a single condition or a set of conditions which a [PointerEvent] has to match
+ * in order to count as an appropriate event for a gesture.
+ *
+ *  Supported matchers:
+ * - [mouse] - will match an event with [PointerType.Mouse] with a required [PointerButton]
+ * - [touch] - will match any event with [PointerType.Touch]
+ * - [stylus] - will match an event with [PointerType.Stylus]. And optional [PointerButton] can be specified.
+ * - [eraser] - will match any event with [PointerType.Eraser]
+ * - [pointer] - takes in [PointerType] and optional [PointerButton]
+ *
+ * Their combination is supported using plus operator:
+ * ```
+ * mouse(PointerButton.Primary) + touch + stylus + eraser
+ * ```
+ * See [DefaultPointerInputMatcher].
+ */
+@ExperimentalFoundationApi
+@OptIn(ExperimentalComposeUiApi::class)
+sealed interface PointerInputMatcher {
```

Comment:
What if make it just `interface` and rename it to `PointerMatcher`?

Reply:
rename to `PointerMatcher ` - Ack

make it just `interface`. 
Pros:
- more flexibility for users (they can override `fun matches(event: PointerEvent): Boolean`)
Cons:
- It will let users provide their own implementations (might be good for them, but...) and we will need to be more careful with this interface in the future. 

if we let users implement it, then we probably need to explain how it's used, so they provide correct implementations. 

for example, one can have such an implementation:

```kotlin
val matchAll = object : PointerMatcher {
   
   override val pointerType: PointerType = PointerType.Unknown
   
   override fun matches(event: PointerEvent) = true
}
```
It will probably work in the most common cases for onClick and onDrag, but might cause some "surprises".

I would wait for potential use cases when such custom implementations are needed. 

Maybe I'm overthinking on this :)